internal/pkg/redis/streams/types.go

```diff
@@ -0,0 +1,110 @@
+package streams
+
+import (
+	"crypto/tls"
+
+	goRedis "github.com/go-redis/redis/v7"
+
+	"github.com/edgexfoundry/go-mod-messaging/pkg/types"
+)
+
+const (
+	// Special identifier used within Redis to signal that a subscriber(consumer) is only interested in the most recent
+	// messages after the client has connected. Redis provides other functionality to read all the data from a stream
+	// even if has been read previously, which is what we want to avoid for functional consistency with the other
+	// implementations of MessageClient.
+	LatestStreamMessage = "$"
+)
+
+// RedisClientCreator type alias for functions which create RedisClient implementation.
+//
+// This is mostly used for testing purposes so that we can easily inject mocks.
+type RedisClientCreator func(redisServerURL string, password string, tlsConfig *tls.Config) (RedisClient, error)
+
+// RedisClient provides functionality needed to read and send messages to/from Redis' RedisStreams functionality.
+//
+// The main reason for this interface is to abstract out the underlying client from Client so that it can be mocked and
+// allow for easy unit testing. Since 'go-redis' does not leverage interfaces and has complicated entities it can become
+// complex to test the operations without requiring a running Redis server.
+type RedisClient interface {
+	// AddToStream sends a message to a stream which acts as a topic.
+	AddToStream(stream string, values map[string]interface{}) error
+	// ReadFromStream blocking operation which retrieves the next message(s) from a stream which acts as a topic.
+	ReadFromStream(stream string) ([]types.MessageEnvelope, error)
+	// Close cleans up any entities which need to be deconstructed.
+	Close() error
+}
+
+// NewGoRedisClientWrapper creates a RedisClient implementation which uses a 'go-redis' Client to achieve the necessary
+// functionality.
+func NewGoRedisClientWrapper(redisServerURL string, password string, tlsConfig *tls.Config) (RedisClient, error) {
```

Comment:
Purely philosophically, I was surprised to see functions in a file called types.go.

Reply:
I agree. I will create a file specifically for this type.